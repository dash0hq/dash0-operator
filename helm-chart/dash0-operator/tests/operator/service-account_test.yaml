suite: test role
templates:
  - operator/service-account.yaml
tests:
  - it: service account should match snapshot (default settings)
    asserts:
      - matchSnapshot: {}

  - it: should not render service account when disabled
    set:
      operator.serviceAccount.create: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should render service account with custom settings
    set:
      operator.serviceAccount.name: custom-service-account-name
    asserts:
      - equal:
          path: metadata.name
          value: custom-service-account-name
