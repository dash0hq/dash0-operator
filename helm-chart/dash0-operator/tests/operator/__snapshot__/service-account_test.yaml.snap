service account should match snapshot (default settings):
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: rbac
        app.kubernetes.io/instance: controller-manager-sa
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: serviceaccount
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-controller-manager
      namespace: NAMESPACE
